here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/AbstractApiTask.java

```diff
@@ -88,34 +86,41 @@ protected AbstractApiTask(
 
   protected <R extends XyzFeature> @NotNull XyzResponse transformReadResultToXyzFeatureResponse(
       final @NotNull Result rdResult, final @NotNull Class<R> type) {
-    return transformResultToXyzFeatureResponse(rdResult, type, NOT_FOUND_ON_NO_ELEMENTS);
+    return transformResultToXyzFeatureResponse(rdResult, type, NOT_FOUND_ON_NO_ELEMENTS, null);
+  }
+
+  protected <R extends XyzFeature> @NotNull XyzResponse transformReadResultToXyzFeatureResponse(
+      final @NotNull Result rdResult, final @NotNull Class<R> type, @Nullable P1<XyzFeature> postProcessing) {
+    return transformResultToXyzFeatureResponse(rdResult, type, NOT_FOUND_ON_NO_ELEMENTS, postProcessing);
```

Comment:
It's not the fault of this PR but I think `AbstractApiTask` could be slightly redesigned before it gets too messy - it clearly starts to violate SRP - there's just too much going on.
If we analyze the reasons for this class to exist at the moment it does a lot:
1)  it validates the `Result`
2) it retrieves features from `Result`
3) it handles missing features
4) it applies postprocessing
5) it send the http response

All of these steps need to be done but I think we will need to refactor this before it gets too hard. This is to be done in separate PR (unless you feel like doing it now).
Just dumping my brain here

Reply:
We can surely create a new separate ticket for addressing issues where we see possibility of improvements. However, the scope of improvement needs to be understood as well, which is atleast by reading the comments, not very clear to me.
I realize it has multiple functions, but what is the proposed way which will make it less complex and not start spreading the code too much?
If the proposal sounds good, we can agree on it and create a ticket in backlog.

Reply:
I see the PR is merged, which is good part.
However, was there a concrete conclusion on this review comment? Or an agreement to not do anything?
@gunplar / @Amaneusz ?

Reply:
@Amaneusz told me he will connect with you some time later to discuss this, so potentially we will have this as a separate task.